src/App.jsx

```diff
@@ -1,23 +1,41 @@
+import React from "react";
 import "./App.css";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+
+// Custom hook to manage semi-persistent state
+function useSemiPersistentState() {
+  // Read from localStorage or default to an empty array
+  const [todoList, setTodoList] = React.useState(
```

Comment:
unless I'm mistaken, you shouldn't have to use 'React.useState'/'React.useEffect', since you imported those individually at the top. Otherwise, good work!